### PR TITLE
feat: adding scrapegraph smartscraper and markdownify for web scraping

### PR DESCRIPTION
**Pull request by PeriniM**
## 🕷️ Add ScrapeGraph Tools for AI Agents Web Scraping

Hey guys nice framework! Added two new tools for webscraping using llm, tried it with `CodeAgent` and works like a charm ;)

This PR adds two new tools that leverage ScrapeGraphAI's services for robust web scraping:

1. `SGSmartScraperTool`: Extracts structured information from webpages using natural language prompts
2. `SGMarkdownifyTool`: Converts webpage content to markdown format with proxy rotation and anti-bot solutions

## Features

- **Proxy Rotation**: Both tools handle complex anti-bot protections automatically
- **Natural Language Interface**: Extract specific information by describing what you want in plain English
- **Structured Output**: Get clean, structured data from any webpage
- **Reliable Scraping**: Works on modern, JavaScript-heavy websites

## Example Usage

Screenshot 2024-12-30 001433
Screenshot 2024-12-30 001445

The example above shows the SmartScraper tool extracting structured information from the Hugging Face Transformers documentation.

## Getting Started

1. Get a free API key from [ScrapeGraphAI Dashboard](https://dashboard.scrapegraphai.com)
2. Set the environment variable:
   ```bash
   SGAI_API_KEY=your_api_key
   ```
3. Install the dependency:
   ```bash
   pip install scrapegraph-py>=1.8.0
   ```

## New Example

Check out `examples/webpage_scraping.py` for a complete example showing how to use these tools with the CodeAgent.

The following is the output of Gradio UI:
Screenshot 2024-12-30 001251

## Dependencies

Added `scrapegraph-py>=1.8.0` to project dependencies in pyproject.toml.

### CODE REVIEW

**Comment by aymeric-roucher** [comment]
Hello @PeriniM,
We prefer to avoid adding too many dependencies to the repo. But you can share your tools to your Hub profile with `tool.push_to_hub(username/tool-path)`!

**Comment by PeriniM** [comment]
> Hello @PeriniM, We prefer to avoid adding too many dependencies to the repo. But you can share your tools to your Hub profile with `tool.push_to_hub(username/tool-path)`!

Sure! I will add it to the hub ;)
